cluster/services/composition.yaml

```diff
@@ -22,9 +22,15 @@ spec:
               # Note that default values are overridden by the patches below.
               name: prometheus
               repository: https://prometheus-community.github.io/helm-charts
-              version: "15.0.1"
+              version: "15.0.2"
             values: {}
       patches:
+        - fromFieldPath: spec.id
+          toFieldPath: metadata.name
+          transforms:
+            - type: string
+              string:
+                fmt: "%s-prometheus"
```

Comment:
i'm not sure if this patch worked as intended, this is the release.helm.crossplane.io that i had when testing just now (note that I don't see `*-prometheus` in its name):
```
> k get managed
NAME                                                              CHART        VERSION   SYNCED   READY   STATE      REVISION   DESCRIPTION        AGE
release.helm.crossplane.io/platform-ref-azure-jared-n6rsv-lch9x   prometheus   15.0.1    True     True    deployed   1          Install complete   25m
```

Reply:
The general patching naming approach did seem to work OK though for other MRs (note `*-rg`, `*-postgresql`, `*-sn`, etc. for example in other names)
```yaml
> k get managed
NAME                                                              CHART        VERSION   SYNCED   READY   STATE      REVISION   DESCRIPTION        AGE
release.helm.crossplane.io/platform-ref-azure-jared-n6rsv-lch9x   prometheus   15.0.1    True     True    deployed   1          Install complete   25m

NAME                                                            READY   SYNCED
resourcegroup.azure.crossplane.io/platform-ref-azure-jared-rg   True    True

NAME                                                                  READY   SYNCED   ENDPOINT                                                  LOCATION    AGE
akscluster.compute.azure.crossplane.io/platform-ref-azure-jared-aks   True    True     platform-ref-azure-jared-7977a8a3.hcp.westus2.azmk8s.io   West US 2   25m

NAME                                                                                READY   SYNCED   VERSION   AGE
postgresqlserver.database.azure.crossplane.io/platform-ref-azure-jared-postgresql   False   False    11        3m42s

NAME                                                                                               READY   SYNCED   STATE   AGE
postgresqlservervirtualnetworkrule.database.azure.crossplane.io/platform-ref-azure-jared-sn-rule   False   False            3m42s

NAME                                                             READY   SYNCED   STATE       AGE
subnet.network.azure.crossplane.io/platform-ref-azure-jared-sn   True    True     Succeeded   25m

NAME                                                                       READY   SYNCED   STATE       LOCATION    AGE
virtualnetwork.network.azure.crossplane.io/platform-ref-azure-jared-vnet   True    True     Succeeded   West US 2   25m
```